csodiaq/loaders/LibraryLoaderStrategyMgf.py

```diff
@@ -0,0 +1,75 @@
+from csodiaq.loaders.LibraryLoaderStrategy import (
+    LibraryLoaderStrategy,
+    create_peaks_from_mz_intensity_lists_and_csodiaq_key_id,
+    remove_low_intensity_peaks_below_max_peak_num,
+    finalVariableNames,
+)
+import pandas as pd
+import os
+from pyteomics import mgf
+import re
+
+
+class LibraryLoaderStrategyMgf(LibraryLoaderStrategy):
+    """
+    Concrete strategy implementation of the LibraryLoaderStrategy strategy class specific for
+        loading mgf library files into a standardized dictionary for use in CsoDIAq.
+
+    Attributes
+    ----------
+    rawLibMgf : pyteomics.mgf.IndexedMGF
+        The original mgf library as loaded using the pyteomics mgf class.
+    """
+
+    def _load_raw_library_object_from_file(self, libraryFilePath: os.PathLike) -> None:
+        self.rawLibMgf = mgf.read(libraryFilePath)
```

Comment:
I might suggest that instead of having `self.rawLibMgf` or `self.rawLibX` that you use `raw_library` or something more generic; that way all the children of LibraryLoaderStragy will have the same attribute for accessing the raw library. We can retain the library type in `self.raw_type` or something, if that info is relevant.

Reply:
My concern with this was that we couldn't standardize the type at the LibraryLoaderStrategy parent class level. Sometimes it's a dataframe, others an mgf object. Would we just make the parent class some broad type, like a dictionary, and then have the children classes change the type?

Reply:
I'll investigate this over the next few days. Functionally it's fine, so I'll merge for now, but it's been added as a task.